### PR TITLE
Remove heapster from project dependencies

### PR DESCRIPTION
**Pull request by MartinForReal**
Remove heapster from project dependencies because it has been retired.

it will make it easier to bump k8s.io dependencies. 

We've found that the client-go library doesn't share transport among different clients. And we may need to bump the dependencies to resolve this issue.

### CODE REVIEW

**Review by mmiranda96 — COMMENTED** [review]
I started with an upgrade effort, but I agree cleaning individual dependencies might be easier. Just a small comment. Thanks for your PR!

File: pkg/exporters/k8sexporter/problemclient/problem_client.go [hunk 1]
```diff
@@ -68,7 +67,7 @@ func NewClientOrDie(npdo *options.NodeProblemDetectorOptions) Client {
 	// we have checked it is a valid URI after command line argument is parsed.:)
 	uri, _ := url.Parse(npdo.ApiServerOverride)
 
-	cfg, err := kubernetes.GetKubeClientConfig(uri)
+	cfg, err := GetKubeClientConfig(uri)
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/exporters/k8sexporter/problemclient/problem_client.go]: updated. Thanks!

**Comment by mmiranda96** [comment]
/lgtm

Thanks for your contribution!

**Pull request merged**